File: daily/python/3066_min_ops_to_exceed_threshold_valueII.py
impl Solution:
	def minOperations(self, nums: List[int], x: int) -> int:
		ret = 0
		heapify(nums)

		while nums[0] < k:
			x = heappop(nums)
			heapreplace(nums, x * 2 + nums[0])
			ret += 1
		return ret

File: daily/cpp/3066_min_ops_to_exceed_threshold_valueII.cpp
#include <bits/stdc++.h>
using namespace std;

using ll = long long;
class Solution {
 public:
  int minOperations(vector<int>& nums, int k) {
    int ans = 0;
    priority_queue<ll, vector<ll>, greater<>> pq;
    for (int x : nums) {
      pq.push((ll)x);
    }
    while (pq.top() < k) {
      ll x = pq.top();
      pq.pop();
      ll y = pq.top();
      pq.pop();
      pq.push(x * 2 + y);
      ans++;
    }

    return ans;
  }
};